### DISCUSSION TITLE
Customizable Icon Colors

### DISCUSSION THREAD

**Discussion by lpnh** [sxyazi/yazi#1144] [2. Show and tell]
I made a simple tool that generates a config file to customize the icon colors using a given color palette:

![before](https://github.com/sxyazi/yazi/assets/90577992/900222f9-dae9-4dd9-8adb-ef4b7eaf5db8) ![after](https://github.com/sxyazi/yazi/assets/90577992/43169be1-2e9b-4a8f-99f0-7db79ce9a3bd)

It's basically just an adaptation of the [tiny-devicons-auto-colors.nvim](https://github.com/rachartier/tiny-devicons-auto-colors.nvim) plugin.

For anyone who enjoys catppuccin's Mocha palette, the config file is already available [here](https://github.com/lpnh/icons-brew.yazi/blob/main/mocha.toml). Simply rename it to `theme.toml` and add it to Yazi config path.

Here's the repo link: https://github.com/lpnh/icons-brew.yazi

**Comment by sxyazi** [comment]
Cool script - never thought we could generate colors based on a palette, and it looks so smooth!

Would you like to raise a PR to add it to https://yazi-rs.github.io/docs/resources so more people can see and use it?

BTW, I noticed it only handles dark colors, is it possible to apply the same process for light colors as well?

**Reply by lpnh** [reply]
> Cool script - never thought we could generate colors based on a palette, and it looks so smooth!

Yeah! But it's more about selecting from a given color set rather than generating a new one. The credit goes entirely to [tiny-devicons-auto-colors.nvim](https://github.com/rachartier/tiny-devicons-auto-colors.nvim)

> BTW, I noticed it only handles dark colors, is it possible to apply the same process for light colors as well?

Done! Now it supports both dark and light colors. Having catppuccin's Latte as default for the light theme.

> Would you like to raise a PR to add it to https://yazi-rs.github.io/docs/resources so more people can see and use it?

Sure! I will happily do so!
